### DISCUSSION TITLE
Fitview to a specific node not working as expected

### DISCUSSION THREAD

**Discussion by navaniharsh31** [xyflow/xyflow#3546] [Q&A]
As soon as reactflow is initialised i want to focus one specific node on screen, but some times it doesnt work and takes the viewport to some random place.

-- Does having negative x or y co ordinates mess up fitView?

```
const handleOnInit = useCallback(
    async (reactFlowInstance: ReactFlowInstance) => {
      setReactFlowInstance(reactFlowInstance);
      const startNode = await getStartNode(bot.nodes);
      reactFlowInstance.fitView({
        nodes: [startNode],
        maxZoom: 0.7,
      });
      if (reactFlowInstance.viewportInitialized) {
        setLoading(false);
      }
    },
    [bot.nodes, getStartNode]
  );
```

**Comment by bcakmakoglu** [comment]
[Mh... works fine for me in a sandbox](https://codesandbox.io/s/gifted-elion-njnsl6?file=/src/Flow.tsx) (though I didn't use `useCallback` - not sure if this *could* be the issue).

Are you using some layouting during this phase as well?

**Reply by navaniharsh31** [reply]
what i assume is happening is when my nodes load, there are a few nodes which have api calls, which eventually increase the height of the nodes. can this be a point of concern for this to not work properly?

**Reply by bcakmakoglu** [reply]
Could very well be, yeah 😄
